spec/unit/webrtc/groupCall.spec.ts

```diff
@@ -14,85 +14,305 @@ See the License for the specific language governing permissions and
+
+        beforeEach(function() {
+            MockRTCPeerConnection.resetInstances();
+
+            client1 = new MockCallMatrixClient(
+                FAKE_USER_ID_1, FAKE_DEVICE_ID_1, FAKE_SESSION_ID_1,
+            ) as unknown as MatrixClient;
+
+            client2 = new MockCallMatrixClient(
+                FAKE_USER_ID_2, FAKE_DEVICE_ID_2, FAKE_SESSION_ID_2,
+            ) as unknown as MatrixClient;
+
+            client1.sendStateEvent = client2.sendStateEvent = (roomId, eventType, content, statekey) => {
+                if (eventType === EventType.GroupCallMemberPrefix) {
+                    const fakeEvent = {
+                        getContent: () => content,
+                        getRoomId: () => FAKE_ROOM_ID,
+                        getStateKey: () => statekey,
+                    } as unknown as MatrixEvent;
+
+                    let subMap = client1Room.currentState.events.get(eventType);
+                    if (!subMap) {
+                        subMap = new Map<string, MatrixEvent>();
+                        client1Room.currentState.events.set(eventType, subMap);
+                        // since we cheat & use the same maps for each, we can
+                        // just add it once.
+                        client2Room.currentState.events.set(eventType, subMap);
+                    }
+                    subMap.set(statekey, fakeEvent);
+
+                    groupCall1.onMemberStateChanged(fakeEvent);
+                    groupCall2.onMemberStateChanged(fakeEvent);
+                }
+                return Promise.resolve(null);
+            };
+
+            const client1Room = new Room(FAKE_ROOM_ID, client1, FAKE_USER_ID_1);
+
+            const client2Room = new Room(FAKE_ROOM_ID, client2, FAKE_USER_ID_2);
+
+            groupCall1 = new GroupCall(
+                client1, client1Room, GroupCallType.Video, false, GroupCallIntent.Prompt, FAKE_CONF_ID,
+            );
+
+            groupCall2 = new GroupCall(
+                client2, client2Room, GroupCallType.Video, false, GroupCallIntent.Prompt, FAKE_CONF_ID,
+            );
+
+            client1Room.currentState.members[FAKE_USER_ID_1] = {
+                userId: FAKE_USER_ID_1,
+            } as unknown as RoomMember;
+            client1Room.currentState.members[FAKE_USER_ID_2] = {
+                userId: FAKE_USER_ID_2,
+            } as unknown as RoomMember;
+
+            client2Room.currentState.members[FAKE_USER_ID_1] = {
+                userId: FAKE_USER_ID_1,
+            } as unknown as RoomMember;
+            client2Room.currentState.members[FAKE_USER_ID_2] = {
+                userId: FAKE_USER_ID_2,
+            } as unknown as RoomMember;
+        });
+
+        afterEach(function() {
+            MockRTCPeerConnection.resetInstances();
+        });
+
+        it("Places a call to a peer", async function() {
+            await groupCall1.create();
+
+            try {
+                const mockSendToDevice = jest.fn();
```

Comment:
Could this get a type?

Reply:
Or, you could do `client1.sendToDevice = jest.fn()` right here and then just reference `client1.sendToDevice` from then on

Reply:
It's kept as its own variable to keep it typed as a mock rather than the plain function type on the client object - did you mean pass the type parameters to the mock function?

Reply:
Yes, that would work too

Reply:
Done